core/domain/app_feedback_report_validators_test.py

```diff
@@ -275,106 +310,142 @@ def test_model_with_web_schema_version_less_than_min_schema_fails(self):
-                    self.COUNTRY_LOCALE_CODE_INDIA),
-                android_device_model=self.ANDROID_DEVICE_MODEL,
-                android_sdk_version=self.ANDROID_SDK_VERSION,
-                entry_point=self.ENTRY_POINT_NAVIGATION_DRAWER,
-                entry_point_topic_id=None, entry_point_story_id=None,
-                entry_point_exploration_id=None, entry_point_subtopic_id=None,
-                text_language_code=self.TEXT_LANGUAGE_CODE_ENGLISH,
-                audio_language_code=self.AUDIO_LANGUAGE_CODE_ENGLISH,
-                android_report_info={},
-                web_report_info=None))
+    def test_model_with_created_on_less_than_earliest_datetime_fails(self):
         model_entity = (
             app_feedback_report_models.AppFeedbackReportModel.get_by_id(
-                entity_id))
-        self.expected_output.append(
+                self.report_id))
+        # Add a scrubber user so that it passes the scrubbing validation.
+        model_entity.scrubbed_by = 'scrubber_user'
+        model_entity.created_on = (
+            app_feedback_report_validators.EARLIEST_VALID_DATETIME -
+            datetime.timedelta(days=4))
+        model_entity.update_timestamps()
+        model_entity.put()
+
+        expected_output = [
             (
-                u'[u\'failed validation check for submitted_on datetime check '
+                u'[u\'failed validation check for created_on datetime check '
                 'of AppFeedbackReportModel\', [u\'Entity id %s: The '
-                'submitted_on field has a value %s which is less than the '
+                'created_on field has a value %s which is less than the '
                 'earliest possible submission date\']]') % (
-                    model_entity.id, model_entity.submitted_on))
-
+                    model_entity.id, model_entity.created_on)]
         self.run_job_and_check_output(
-            self.expected_output, sort=True, literal_eval=False)
+            expected_output, sort=False, literal_eval=False)
 
-    def test_validation_on_expired_model_without_scrubbed_by_field_fails(self):
+    def test_model_created_on_inside_buffer_no_scrubbed_by_field_is_valid(
+            self):
+        valid_datetime_buffer = (
+            app_feedback_report_validators.VALID_SCRUBBING_DATETIME_BUFFER -
+            datetime.timedelta(days=1))
+        valid_datetime = datetime.datetime.utcnow() - (
+            feconf.APP_FEEDBACK_REPORT_MAXIMUM_DAYS + valid_datetime_buffer)
         model_entity = (
             app_feedback_report_models.AppFeedbackReportModel.get_by_id(
                 self.report_id))
-        model_entity.created_on = (
-            datetime.datetime.utcnow() - datetime.timedelta(days=100))
+        model_entity.created_on = valid_datetime
         model_entity.update_timestamps()
         model_entity.put()
-        output = [
+
+        # Swap the created_on datetime validation check to suppress the output
```

Comment:
Ah I should name this better -- this tests the scrubbed_by field but changing the created_on datetime adds an error that the created_on datetime is before the earliest valid datetime.

The error is because the scrubbed_by validation uses the current datetime to determine what should have a scrubbed_by field, so the created_on datetime will be validated against 90 days prior to the current time the test is run. This test changes it to be 91 days prior to the current time so this will fail the created_on datetime check so long as 91 days before the current datetime is earlier than the earliest valid datetime of a submitted report (the first 90 days that this test will be included in code.

An alternative is to set the earliest valid datetime to be more than 90 days earlier from now (i.e. January 2021) but that's not necessarily accurate as the earliest possible date we might receive a report.